frontend/tests/controllers/RunDetailsTest.php

```diff
@@ -361,4 +361,97 @@ public function testUserInContest(
             $acRunData['details']['runs'][0]['contest_alias']
         );
     }
+
+    /**
+     * A PHPUnit data provider for the test with valid show_diff values.
+     *
+     * @return list<list<string>>
+     */
+    public function showDiffValueProvider(): array {
+        $expectedCases = [
+            'none' => [],
+            'examples' => [
+                'sample' => [
+                    'in' => "1 2\n",
+                    'out' => "3\n",
+                ],
+            ],
+            'all' => [
+                'easy.00' => [
+                    'in' => "1 2\n",
+                    'out' => "3\n",
+                ],
+                'easy.01' => [
+                    'in' => "2 3\n",
+                    'out' => "5\n",
+                ],
+                'medium.00' => [
+                    'in' => "100 200\n",
+                    'out' => "300\n",
+                ],
+                'medium.01' => [
+                    'in' => "1234 5678\n",
+                    'out' => "6912\n",
+                ],
+                'sample' => [
+                    'in' => "1 2\n",
+                    'out' => "3\n",
+                ],
+            ],
+        ];
+        return [
+            ['none', $expectedCases['none']],
+            ['examples', $expectedCases['examples']],
+            ['all', $expectedCases['all']],
+        ];
+    }
+
+    /**
+     * @param array{all: array<string, array{in: string, out: string}, examples: array<string, array{in: string, out: string}, none: array<string, array{in: string, out: string}}
```

Comment:
aquí no se está pasando el arreglo completo, sino sólo un elemento, entonces esto debería ser:

```suggestion
     * @param array<string, array<string, string>> $cases
```

(también, faltaba el `$cases`).

Reply:
Cierto. Se arregla.